app/domain/authentication/authn_iam/authenticator.rb

```diff
@@ -76,6 +97,25 @@ def response_from_signed_request(aws_headers)
           body.dig('ErrorResponse', 'Error', 'Message').to_s.strip
         )
       end
+
+      # Extracts the STS region from the host header if it exists.
+      # If not, we use the authorization header's credential string, i.e.:
+      # Credential=AKIAIOSFODNN7EXAMPLE/20220830/us-east-1/sts/aws4_request
+      def extract_sts_region(signed_headers)
+        host = signed_headers['host']
+
+        if host == 'sts.amazonaws.com'
+          return 'global'
+        end
+      
+        match = host&.match(%r{sts.([\w\-]+).amazonaws.com})
+        return match&.captures&.first if match
```

Comment:
Looking at this again, and even given my suggestion:
```ruby
return match.captures&.first if match
```

I wonder if we still need this Safe Navigation Operator.
- [`.match`](https://ruby-doc.org/core-3.1.1/String.html#method-i-match) returns `nil` if there are no matches found
- [`.match`](https://ruby-doc.org/core-3.1.1/String.html#method-i-match) returns a [`MatchData` object](https://ruby-doc.org/core-3.1.1/MatchData.html) for 1+ matches found

That means:
- `if match` catches cases with no matches
- Otherwise, there is guaranteed to be at least one match - `match.captures` will not be `nil`.

We might be able to do this instead, with no change in behavior and less `&`s:
```ruby
return match.captures.first if match
```